src/overlays/actors/ovl_En_Niw_Girl/z_en_niw_girl.c

```diff
@@ -21,21 +31,226 @@ const ActorInit En_Niw_Girl_InitVars = {
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ きゃははははは、まてー ☆☆☆☆☆ %d\n" VT_RST, this->unkFlag);
+        osSyncPrintf("\n\n");
+        thisx->colChkInfo.mass = 0xFF;
+        this->actionFunc = EnNiwGirl_Talk;
+    } else {
+        osSyncPrintf("\n\n");
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ なぜか、セットできむぅあせん ☆☆☆☆☆ %d\n" VT_RST, thisx->params);
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ んんがくく ☆☆☆☆☆ %d\n" VT_RST, this->unkFlag);
+        osSyncPrintf("\n\n");
+        Actor_Kill(thisx);
+    }
+}
+
+void EnNiwGirl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnNiwGirl_PlayJumpAnimation(EnNiwGirl* this, GlobalContext* globalCtx) {
+    f32 tempFrames = SkelAnime_GetFrameCount(&D_06000378);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000378, 1.0f, 0.0f, tempFrames, 0, -10.0f);
+    this->actor.flags &= ~1;
+    this->actionFunc = func_80AB9210;
+}
+
+void func_80AB9210(EnNiwGirl* this, GlobalContext* globalCtx) {
+    Path* path = &globalCtx->setupPathList[this->unkFlag];
+    f32 xDistBetween;
+    f32 zDistBetween;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_SmoothScaleMaxF(&this->actor.speedXZ, 3.0f, 0.2f, 0.4f);
+
+    // Find the X and Z distance between the girl and the cuckoo she is chasing
+    xDistBetween = this->attachedActor->actor.posRot.pos.x - this->actor.posRot.pos.x;
+    zDistBetween = this->attachedActor->actor.posRot.pos.z - this->actor.posRot.pos.z;
+    if (func_8010BDBC(&globalCtx->msgCtx) != 0) {
+        this->attachedActor->unk_2E8 = 0;
+    }
+    if (sqrtf(SQ(xDistBetween) + SQ(zDistBetween)) < 70.0f) {
+        this->attachedActor->unk_2E8 = (this->unkFlag + 1);
+        this->attachedActor->unk_2EC = path->count;
+    } else if (sqrtf(SQ(xDistBetween) + SQ(zDistBetween)) > 150.0f) {
+        this->attachedActor->unk_2E8 = 0;
+    }
+
+    // Change her angle so that she is always facing the cuckoo
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, Math_atan2f(xDistBetween, zDistBetween) * 10430.378f, 3,
+                            this->unk_27C, 0);
+    Math_SmoothScaleMaxF(&this->unk_27C, 5000.0f, 30.0f, 150.0f);
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+
+    //*Only allow Link to talk to her when she is playing the jumping animation
+    if ((this->jumpTimer == 0) || (func_8008F080(globalCtx) != 0)) {
+        this->jumpTimer = 60;
+        this->actionFunc = EnNiwGirl_Talk;
+    }
+}
+
+void EnNiwGirl_Talk(EnNiwGirl* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06009C78, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06009C78), 0, -10.0f);
+    this->actor.flags |= 1;
+    this->actor.textId = 0x7000; // I can't catch that Cucco!
+    if ((gSaveContext.eventChkInf[8] & 1) && (this->unk_27A == 0)) {
+        this->actor.textId = 0x70EA; // That scary-looking man took off out of here really fast !
+    }
+    switch (func_8008F080(globalCtx) - 1) {
```

Comment:
This switch should probably be on the return value of the function (without - 1), with all cases incremented by 1 to match.
```suggestion
    switch (func_8008F080(globalCtx)) {
```